crates/mun_codegen_macros/src/lib.rs

```diff
@@ -27,175 +21,189 @@ pub fn as_value_derive(input: TokenStream) -> TokenStream {
+                        // same variable name.
+                        let type_context = context;
+
+                        let field_types = vec![ #(#field_types),* ];
+                        let field_padding = #field_padding_types;
+                        let struct_fields: Vec<_> = field_padding
+                            .into_iter()
+                            .map(|p| {
+                                let (ty, num_chunks) = if p % 8 == 0 {
+                                    (context.context.i64_type(), p / 8)
+                                } else if p % 4 == 0 {
+                                    (context.context.i32_type(), p / 4)
+                                } else if p % 2 == 0 {
+                                    (context.context.i16_type(), p / 2)
+                                } else {
+                                    (context.context.i8_type(), p)
+                                };
+
+                                ty.array_type(num_chunks as u32).into()
+                            })
+                            .interleave(field_types.into_iter())
+                            .collect();
+
+                        struct_ty.set_body(&struct_fields, true);
+                        struct_ty
+                    }
+                }
 
-        impl<'ink> crate::value::PointerValueType<'ink> for #ident {
-            fn get_ptr_type(context: &crate::value::IrTypeContext<'ink, '_>, address_space: Option<inkwell::AddressSpace>) -> inkwell::types::PointerType<'ink> {
-                Self::get_ir_type(context).ptr_type(address_space.unwrap_or(inkwell::AddressSpace::Generic))
-            }
-        }
+                impl<'ink> crate::value::PointerValueType<'ink> for #ident {
+                    fn get_ptr_type(context: &crate::value::IrTypeContext<'ink, '_>, address_space: Option<inkwell::AddressSpace>) -> inkwell::types::PointerType<'ink> {
+                        Self::get_ir_type(context).ptr_type(address_space.unwrap_or(inkwell::AddressSpace::Generic))
+                    }
+                }
+
+                impl<'ink> crate::value::AsValue<'ink, #ident> for #ident {
+                    fn as_value(&self, context: &crate::value::IrValueContext<'ink, '_, '_>) -> crate::value::Value<'ink, Self> {
+                        fn padded_size(align: usize, data_size: usize) -> usize {
+                            ((data_size + align - 1) / align) * align
+                        }
+
+                        // Aliasing to make sure that all procedurally generated macros can use the
+                        // same variable name.
+                        let type_context = context.type_context;
+                        let field_padding = #field_padding_values;
+
+                        let struct_type = Self::get_ir_type(context.type_context);
+                        // eprintln!("Constructing: {:?}", struct_type.print_to_string().to_string());
```

Comment:
Can this be removed?

Reply:
I can remove them. The code existed before my changes, so I thought they were there for a purpose.

Reply:
Ah yes I remember that I put it there to be able to debug the code.